src/plugins/usage_collection/server/routes/stats.ts

```diff
@@ -0,0 +1,204 @@
+ */
+
+import { schema } from '@kbn/config-schema';
+import { i18n } from '@kbn/i18n';
+import defaultsDeep from 'lodash/defaultsDeep';
+import { Observable } from 'rxjs';
+import { first } from 'rxjs/operators';
+
+import {
+  AuthStatus,
+  GetAuthState,
+  IRouter,
+  LegacyAPICaller,
+  MetricsServiceSetup,
+  ServiceStatus,
+  ServiceStatusLevels,
+} from '../../../../core/server';
+import { CollectorSet } from '../collector';
+
+const STATS_NOT_READY_MESSAGE = i18n.translate('usageCollection.stats.notReadyMessage', {
+  defaultMessage: 'Stats are not ready yet. Please try again later.',
+});
+
+const SNAPSHOT_REGEX = /-snapshot/i;
+
+export function registerStatsRoute({
+  router,
+  config,
+  collectorSet,
+  metrics,
+  overallStatus$,
+  getAuthState,
+}: {
+  router: IRouter;
+  config: {
+    allowAnonymous: boolean;
+    kibanaIndex: string;
+    kibanaVersion: string;
+    uuid: string;
+    server: {
+      name: string;
+      hostname: string;
+      port: number;
+    };
+  };
+  collectorSet: CollectorSet;
+  metrics: MetricsServiceSetup;
+  overallStatus$: Observable<ServiceStatus>;
+  getAuthState: GetAuthState;
+}) {
+  const getUsage = async (callCluster: LegacyAPICaller): Promise<any> => {
+    const usage = await collectorSet.bulkFetchUsage(callCluster);
+    return collectorSet.toObject(usage);
+  };
+
+  const getClusterUuid = async (callCluster: LegacyAPICaller): Promise<string> => {
+    const { cluster_uuid: uuid } = await callCluster('info', { filterPath: 'cluster_uuid' });
+    return uuid;
+  };
+
+  router.get(
+    {
+      path: '/api/stats',
+      options: {
+        // Even when allowAnonymous is true, we attempt to authenticate in order to deny anonymous users access to
+        // extended stats.
+        authRequired: config.allowAnonymous ? 'optional' : true,
+        tags: ['api'], // ensures that unauthenticated calls receive a 401 rather than a 302 redirect to login page
+      },
+      validate: {
+        query: schema.object({
+          extended: schema.maybe(
+            schema.oneOf([schema.literal(''), schema.literal('true'), schema.literal('false')])
+          ),
+          legacy: schema.maybe(
+            schema.oneOf([schema.literal(''), schema.literal('true'), schema.literal('false')])
+          ),
+          exclude_usage: schema.maybe(
+            schema.oneOf([schema.literal(''), schema.literal('true'), schema.literal('false')])
+          ),
+        }),
+      },
+    },
+    async (context, req, res) => {
+      const isExtended = req.query.extended === '' || req.query.extended === 'true';
+      const isLegacy = req.query.legacy === '' || req.query.legacy === 'true';
+      const shouldGetUsage =
+        req.query.exclude_usage === undefined || req.query.exclude_usage === 'false';
+
+      let extended;
+      if (isExtended) {
+        // Unauthenticated users should never have access to these stats.
+        if (getAuthState(req).status === AuthStatus.unauthenticated) {
```

Comment:
I was coding to directly to the tests it seems...though having this check doesn't change the behavior, the 401 behavior when `status.allowAnonymous = true` is set happens implicitly due to the ES API calls with the current user's credentials. I'll remove this explicit check since it is unnecessary.